examples/expdesign.jl

```diff
@@ -81,16 +81,78 @@ dOpt = maximize(
 # @show b
 # @show cones
 
-solve!(dOpt, (enforce_integrality ? solver : cont_solver))
-println("  objective $(dOpt.optval)")
+solve!(dOpt, solver)
+println("\n  objective $(dOpt.optval)")
 println("  solution\n$(np.value)")
 
+# JuMP.jl
+# MI-SOC-SDP reformulation of D-optimal design
+#   maximize    nthroot det V*diag(lambda)*V'
+#   subject to  sum(lambda)=1,  lambda >=0
+println("\n\n****D optimal: JuMP.jl****\n")
+
+function eigenvals(dOpt, A::Array{JuMP.Variable,2})
+    dimA = size(A,1)
+    U = @variable(dOpt, [i=1:dimA, j=i:dimA])
+    for i in 1:dimA
+        setlowerbound(U[i,i], 0)
+    end
+    # @SDconstraint(dOpt, A >= 0) # do we want this? A = V * diagm(x./n) * V' is PSD automatically if x >= 0
+    Umat = AffExpr[((j < i) ? 0 : U[i,j]) for i=1:dimA, j=1:dimA]
+    @SDconstraint(dOpt, [diagm([U[i,i] for i in 1:dimA]) Umat; Umat' A] >= 0)
+    return [U[i,i] for i in 1:dimA]
+end
+
+function scaledGeomean(dOpt, x::Vector{JuMP.Variable})
+    dimx = length(x)
+    if dimx > 2
+        dimxbar = Int(2^ceil(log(2, dimx)))
+        half_dimxbar = Int(dimxbar / 2)
+        first_half = x[1:half_dimxbar]
+        xone = @variable(dOpt, [1:(dimxbar - dimx)], lowerbound=1, upperbound=1)
+        last_half = vcat(vec(x[(half_dimxbar + 1):end]), xone) # append ones to last half until it's a power of 2
+        return geomean(dOpt, scaledGeomean(dOpt, first_half), scaledGeomean(dOpt, last_half))
+    elseif dimx == 2
+        return geomean(dOpt, x[1], x[2])
+    else
+        return x
+    end
+end
+
+function geomean(dOpt, x::JuMP.Variable, y::JuMP.Variable) # SOCRotated
+    t = @variable(dOpt, lowerbound=0)
+    @constraint(dOpt, x*y >= t^2)
+    return t
+end
+
+dOpt = Model(solver=solver)
+np = @variable(dOpt, [j=1:p], Int, lowerbound=0, upperbound=nmax)
+@constraint(dOpt, sum(np) <= n)
+Q = @variable(dOpt, [1:q,1:q])
```

Comment:
OK does this reduce the number of variables then? rather than enforcing symmetry